src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -3510,15 +3522,16 @@ void MacroAssembler::cmpxchg_weak(Register addr, Register expected,
                                   enum operand_size size,
                                   Assembler::Aqrl acquire, Assembler::Aqrl release,
                                   Register result) {
-  if (UseZacas) {
-    cmpxchg(addr, expected, new_val, size, acquire, release, result, true);
-    return;
-  }
 
   assert_different_registers(addr, t0);
   assert_different_registers(expected, t0);
   assert_different_registers(new_val, t0);
 
+  if (UseZacas) {
+    cmpxchg(addr, expected, new_val, size, acquire, release, result, true);
+    return;
+  }
```

Comment:
Question: Should we move the code after the three assertions? I see same assertions are there in `MacroAssembler::cmpxchg` as well. Even though they are the same for now, I assume they target different code and are submit to change in the future.

Reply:
The reason to move it after is readability.
By having the zacas code before the asserts you think the implied contract the asserts states do not apply to zacas case. Meaning that, at least I, read the code as addr, expected, new_val may be t0 when UseZacas is true.

Reply:
Make sense in that respect.